### PR TITLE
Add ReflectometryISISCalibration algorithm

### PR DESCRIPTION
**Pull request by rbauststfc**
**Description of work.**

This PR is the first step required for issue #32022. It adds the new calibration algorithm required to support 2D detectors. The algorithm corrects the Y positions of the detectors in the input workspace using the positions provided in a calibration file. This algorithm is still a work in progress - we have assumed that the positions of all the detectors included in the calibration file should be corrected, however we are waiting on confirmation from the scientists as to whether it should be possible to apply only a subset of the corrections from the calibration file.

At the moment the algorithm is creating a CalibTable workspace in the ADS to give us visibility over the calibration values being applied. In this PR a hard-coded output name is used, meaning that if this algorithm was run multiple times for different input workspaces you would get only one CalibTable that was overwritten by each subsequent run. It seems more sensible for the CalibTable workspace to be created with a unique name instead. As this is still a work in progress, that will be changed in a future PR as it becomes even more of a consideration once we're calling this algorithm from our existing reduction algorithms.

Support for workspace groups will be added in a future PR, if needed.

**To test:**

1) For now it's probably sufficient that the unit tests pass.
2) The reviewer could also create the workspace given in the example script on the linked issue and run the `ReflectometryISISCalibration` algorithm from Workbench using the calibration file in the example. The algorithm will not currently give the same output as running the script (it will calibrate a larger range of detectors), however it should still complete without error.

Refs #32022. 

*This does not require release notes* because release notes will be added in a future PR that implements more of the functionality required for the issue.

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/ReflectometryISISCalibration.py [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+                        'Detector Width': box[0] * scalings[0],
+                        'Detector Height': box[1] * scalings[1]}
+            table.addRow(next_row)
+        return table
+
+    def _calculate_calibrated_y_pos(self, specpixel_y, scanned_specpixel_y, scanned_pixel_pos):
+        # First find the difference between the specular pixel position and the pixel position in the scan data.
+        # The calibrated Y co-ordinate is then found by adding this difference to the specular pixel position from
+        # the IDF.
+        # Need to confirm what the multiply 0.001 is for.
```
Comment (gemmaguest) [inline Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/ReflectometryISISCalibration.py:147]: I think the input file must be in `mm` so this must be the conversion to `m`. It's probably worth mentioning the expected units in the doc string for the input file.

**Review by gemmaguest — CHANGES_REQUESTED** [review]
Code looks really good :+1: :+1: Just one minor suggestion to update a doc string and comment.

File: Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/ReflectometryISISCalibration.py (hunk 1 above)
Comment (rbauststfc) [reply Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/ReflectometryISISCalibration.py:147]: Ah, thanks, that's a good spot! I'll update the doc/comments.

**Review by gemmaguest — APPROVED** [review]
Thanks for the changes - looks good now :+1: Functional testing also looks good - I compared it to the script in the issue and the results look like they are the same for the relevant spectra :+1:

**Pull request merged**